ebean-core/src/main/java/io/ebeaninternal/server/type/DefaultTypeManager.java

```diff
@@ -404,7 +405,11 @@ private ScalarType<?> createJsonObjectMapperType(DeployProperty prop, int dbType
     if (MutationDetection.DEFAULT == prop.getMutationDetection()) {
       prop.setMutationDetection(jsonManager.mutationDetection());
     }
-    var req = new ScalarJsonRequest(jsonManager, dbType, docType, prop.getOwnerType(), prop.getMutationDetection(), prop.getName());
+    Class<?> type = prop.getOwnerType();
+    if (prop instanceof DeployBeanProperty) {
+      type = ((DeployBeanProperty) prop).getField().getDeclaringClass();
```

Comment:
Würde es Sinn machen, den Code in "getOwnerType" zu verlagern? Oder geht dann zu viel kaputt?